pkg/snapshot/copier/copier_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package copier_test
+
+import (
+	"os"
+	"reflect"
+
+	"github.com/gardener/etcd-backup-restore/pkg/miscellaneous"
+	. "github.com/gardener/etcd-backup-restore/pkg/snapshot/copier"
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	copierConfig    *brtypes.CopierConfig
+	snapstoreConfig *brtypes.SnapstoreConfig
+	ss              brtypes.SnapStore
+	ds              brtypes.SnapStore
+	copier          *Copier
+)
+
+var _ = Describe("Running Copier", func() {
+	BeforeEach(func() {
+		copierConfig = NewCopierConfig()
+		copierConfig.SourceSnapstore = &brtypes.SnapstoreConfig{
+			MaxParallelChunkUploads: 5,
+			TempDir:                 "/tmp",
+			Provider:                "Local",
+			Container:               targetSnapstoreDir,
+		}
+		snapstoreConfig = &brtypes.SnapstoreConfig{
+			MaxParallelChunkUploads: 5,
+			TempDir:                 "/tmp",
+			Provider:                "Local",
+			Container:               snapstoreDir,
+		}
+
+		var err error
+		ds, ss, err = GetSourceAndDestinationStores(copierConfig, snapstoreConfig)
+		Expect(err).ToNot(HaveOccurred())
+		copier = NewCopier(ss, ds, logger)
+	})
+	It("should run the copy command succesfully", func() {
+		fullSourceStoreSnapshot, deltaSrourceStoreSnapList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(ss)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(copier.Run()).ToNot(HaveOccurred())
+		fullTargetStoreSnapshot, deltaTargetStoreSnapList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(ds)
+		Expect(snapsAreTheSame(fullSourceStoreSnapshot, fullTargetStoreSnapshot)).To(BeTrue())
+		for i, ssnap := range deltaSrourceStoreSnapList {
+			Expect(len(deltaSrourceStoreSnapList)).To(Equal(len(deltaTargetStoreSnapList)))
+			Expect(snapsAreTheSame(ssnap, deltaTargetStoreSnapList[i])).To(BeTrue())
+		}
+		_, err = os.Stat(targetSnapstoreDir)
+		Expect(err).ToNot(HaveOccurred())
+		os.Remove(targetSnapstoreDir)
```

Comment:
Perhaps I am missing something, but should't these lines be part of an AfterEach call. Also if the file does not exist, the tests will fail.

Reply:
Its a leftover it is already called [here](https://github.com/gardener/etcd-backup-restore/blob/512a4c986f68c0f42b54e390ff50a728c8abc260/pkg/snapshot/copier/copier_suite_test.go#L100)